### ISSUE TITLE
Wallpaper Changer applet - Feedback/Bug reports

### ISSUE THREAD

**Issue by Odyseus** [Odyseus/CinnamonTools#42]
## Applet status

- Applet ready to be published. Estimated date of publishing: **May 29, 2017**.
- Applet ready to be translated.

## Description

Wallpaper Changer is an applet based on the gnome-shell extension called [Desk Changer](https://github.com/BigE/desk-changer) by [Eric Gach](https://github.com/BigE). A wallpaper slideshow applet with multiple profiles support.

## Tested environments

* [x] ![Cinnamon 3.0](https://odyseus.github.io/CinnamonTools/lib/badges/cinn-3.0.svg)
* [x] ![Cinnamon 3.2](https://odyseus.github.io/CinnamonTools/lib/badges/cinn-3.2.svg)
* [x] ![Cinnamon 3.4](https://odyseus.github.io/CinnamonTools/lib/badges/cinn-3.4.svg)

<span style="color:red;font-size:large;">
**Important note:** Do not try to install and force compatibility for any other version of Cinnamon older than 3.0.x. As a protection mechanism, the applet will auto-remove itself from the panel.
</span>

## Features

- Possibility to create and switch between several profiles. A profile is simply a list of images and/or folders containing images that this applet will use to switch the wallpaper.
- Possibility to preview the next wallpaper from this applet menu.
- Wallpapers can be switched on demand from the controls found in this applet menu.
- The wallpapers rotation can be alphabetically or random.
- The wallpapers rotation can be defined by an interval in seconds or hourly.
- Possibility to open the next or current wallpapers from this applet menu.
- Possibility to display a notification every time the wallpaper is switched.
- Configurable hotkeys to switch to next/previous wallpaper.
- Read the tooltips of each option on this applet settings window for more details.

**Note:** This applet doesn't complement the Cinnamon option called **Play backgrounds as a slideshow**, it replaces it. The Cinnamon option should be disabled at all times for this applet to work as expected. No worries, nothing *fatal* could happen.

## Images

![screenshot](https://cloud.githubusercontent.com/assets/3822556/25260821/fdf7fefa-2624-11e7-8c25-52be83686015.png)

## Known issues

No known issues for the moment.

## Issue reports

**Issue reporters should adjunct the output of the following commands.**
**Check the content of the log files for sensible information BEFORE running the commands!!!**

`inxi -xxxSc0 -! 31`
`pastebin ~/.cinnamon/glass.log`
`pastebin ~/.xsession-errors`

## References to anyone that could be interested in testing the applet

@buzz @copecu @fortalezense @maladro1t @NikoKrause @Radek71 @sphh @DamienNZ @muzena @eson57 @giwhub

**Comment by Radek71** [comment]
@Odyseus 

- please add the string "default" to .pot file.
- idea for improvement: change the icons/symbols for alphabetical order (something with letter A-Z for example) and for timed rotation (something with clock symbol)

radek

**Comment by Odyseus** [comment]
Hello, @Radek71.

Sadly, the "default" string cannot be translated. It isn't just the name of the default profile, it's also the value that is used to programmatically identify the default profile. Maybe I can simply change the name of the default profile to Cinnamon. That way it wouldn't need translation. What do you think?

About the icons. I really suck at choosing icons and that's why I left the original ones for those controls. I think that they represent accurately enough the currently active actions.

> change the icons/symbols for alphabetical order (something with letter A-Z for example)...

There are two *problems* with that. First, that the alphabetical order isn't strictly alphabetical. It first shows the files that start with upper case and then the ones that start with lower case. This is an annoyance that I don't know how to *fix*. And second, using an image that has the letters A-Z isn't very *localization friendly*.

> and for timed rotation (something with clock symbol)

That button has three states (**Hourly** - **Interval** - **Stopped**). The **Hourly** state already has an icon that looks like a clock (at lest, that's how it looks like with the icon theme that I use).

**Comment by anon4259** [comment]
I've switched to MATE on my main machine so you no longer have to ping me in the testing threads, but if I switch back to Cinnamon some day I will be sure to use your applets because they are top notch.

**Comment by pizzadude** [comment]
I've switched to MATE on my main machine so you no longer have to ping me in the testing threads, but if I switch back to Cinnamon some day I will be sure to use your applets because they are top notch.

**Comment by Odyseus** [comment]
Thanks for your words, @pizzadude. Enjoy Mate. 👍

**Comment by Odyseus** [comment]
### Second prototype changelog
- Added missing localizations.sh script.
- Changed the name of the default profile to Cinnamon.
- Some code clean up.

**Comment by NikoKrause** [comment]
@Odyseus 
How to translate HELP.md?
Just creating HELP_de.md and translate it?

**Comment by Odyseus** [comment]
Hello, @NikoKrause.

Sadly, yes. That's the best that I could do for the moment. I implemented something similar to what you proposed [here](https://github.com/Odyseus/CinnamonTools/issues/19#issuecomment-274101427), but without the need to open different HTML files (those individual HELP-lgn.md files are all unified into one HTML file and the localization can be selected with buttons). The initial translation isn't that hard. But updating them is the most annoying thing of all.

I'm still looking for a method that would allow me to extract English strings and insert them into a .pot file so I can then create localized HTML files, but all methods that I could find have some flaw or another. In short, HTML files are a nightmare to localize. BTW, I already looked at some of the TranslateHouse tools and some of them gave me some ideas that I will try.
